core/config/project_settings.cpp

```diff
@@ -412,6 +412,43 @@ void ProjectSettings::_convert_to_last_version(int p_from_version) {
 	}
 }
 
+bool ProjectSettings::_attempt_load_from_separate_pack(const String &p_exec_path) {
+	String exec_dir = p_exec_path.get_base_dir();
+	String exec_filename = p_exec_path.get_file();
+	// This code continues stripping the file name of extensions until
+	// it finds a PCK, or until there are no more extensions to strip.
+	// For example, let's say our executable is "mygame.x86_64.exe".
+	// First we look for "mygame.x86_64.exe.pck". If that fails, we look
+	// for "mygame.x86_64.pck". If that fails, we look for "mygame.pck".
+	while (true) {
+#ifdef OSX_ENABLED
+		// Attempt to load PCK from macOS .app bundle resources.
+		if (_load_resource_pack(OS::get_singleton()->get_bundle_resource_dir().plus_file(exec_filename + ".pck"))) {
+			return true;
+		}
+#endif
```

Comment:
The implementation in this PR has the goal of being as simple as possible while allowing these extra use cases. Indeed this will still work fine for macOS. In fact, if the executable is extensionless, this should end up loading on the first try, skipping all of the other logic that would be pointless.

Plus this adds flexibility in the event that someone wanted to not use .app bundles. Even if nobody uses that... it doesn't matter, we don't need to use every possibility of a feature to have a flexible solution be worth it.